modules/plottinggl/plotters/persistencediagramplotgl.cpp

```diff
@@ -0,0 +1,629 @@
+}
+
+void PersistenceDiagramPlotGL::renderLines(const size2_t &dims,
+                                           const std::vector<uint32_t> &diagonalIndices,
+                                           const std::vector<uint32_t> &indices) {
+    lineShader_.setUniform("roundCaps", true);
+    lineShader_.setUniform("screenDim", vec2(dims));
+    if (!diagonalIndices.empty()) {
+        lineShader_.setUniform("lineWidth", properties_.lineWidthDiagonal_.get());
+        lineShader_.setUniform("pickingEnabled", false);
+
+        glDrawElements(GL_LINES, static_cast<uint32_t>(diagonalIndices.size()), GL_UNSIGNED_INT,
+                       diagonalIndices.data());
+    }
+
+    if (!indices.empty()) {
+        lineShader_.setUniform("lineWidth", properties_.lineWidth_.get());
+        lineShader_.setUniform("pickingEnabled", true);
+
+        glDrawElements(GL_LINES, static_cast<uint32_t>(indices.size()), GL_UNSIGNED_INT,
+                       indices.data());
+    }
+}
+
+void PersistenceDiagramPlotGL::renderPoints(const size2_t &dims,
+                                            const std::vector<uint32_t> &indices) {
+    pointShader_.setUniform("pixelSize", vec2(1.0f) / vec2(dims));
+    pointShader_.setUniform("dims", ivec2(dims));
+    pointShader_.setUniform("maxRadius", properties_.radius_.get());
+    pointShader_.setUniform("borderWidth", properties_.borderWidth_.get());
+    pointShader_.setUniform("borderColor", properties_.borderColor_.get());
+
+    glDrawElements(GL_POINTS, static_cast<uint32_t>(indices.size()), GL_UNSIGNED_INT,
+                   indices.data());
+}
+
+void PersistenceDiagramPlotGL::setXAxisLabel(const std::string &label) {
+    properties_.xAxis_.setTitle(label);
+}
+
+void PersistenceDiagramPlotGL::setYAxisLabel(const std::string &label) {
+    properties_.yAxis_.setTitle(label);
+}
+
+void PersistenceDiagramPlotGL::setXAxis(std::shared_ptr<const plot::Column> col) {
+    setXAxisLabel(col->getHeader());
+    setXAxisData(col->getBuffer());
+}
+
+void PersistenceDiagramPlotGL::setYAxis(std::shared_ptr<const plot::Column> col) {
+    setYAxisLabel(col->getHeader());
+    setYAxisData(col->getBuffer());
+}
+
+void PersistenceDiagramPlotGL::setXAxisData(std::shared_ptr<const BufferBase> buffer) {
+    xAxis_ = buffer;
+    if (buffer) {
+        auto minmax = util::bufferMinMax(buffer.get(), IgnoreSpecialValues::Yes);
+        minmaxX_.x = static_cast<float>(minmax.first.x);
+        minmaxX_.y = static_cast<float>(minmax.second.x);
+
+        properties_.xAxis_.setRange(minmaxX_);
+
+        // make sure that vertical axis has same minimum
+        minmaxY_.x = minmaxX_.x;
+        properties_.yAxis_.setRange(minmaxX_);
+    }
+}
+
+void PersistenceDiagramPlotGL::setYAxisData(std::shared_ptr<const BufferBase> buffer) {
+    yAxis_ = buffer;
+    if (buffer) {
+        auto minmax = util::bufferMinMax(buffer.get(), IgnoreSpecialValues::Yes);
+        // minmaxY_.x = static_cast<float>(minmax.first.x);
```

Comment:
Why comment?

Reply:
left-over from before when `minmaxY_.x` was not set in `setXAxisData()`